libs/langchain/langchain/utilities/tensorflow_datasets.py

```diff
@@ -0,0 +1,108 @@
+    To get started see the Guide: https://www.tensorflow.org/datasets/overview and
+    the list of datasets: https://www.tensorflow.org/datasets/catalog/overview#all_datasets
+
+    You have to provide the sample_to_document_function: a function that converts a sample
+     from the dataset-specific format to the Document.
+
+    Args:
+        dataset_name: the name of the dataset to load
+        split_name: the name of the split to load. Defaults to "test".
+        load_max_docs: a limit to the number of loaded documents. Defaults to 100.
+        sample_to_document_function: a function that converts a dataset sample to a Document
+
+    Example:
+        .. code-block:: python
+
+            from langchain.utilities import TensorflowDatasets
+
+            def mlqaen_example_to_document(example: dict) -> Document:
+                return Document(
+                    page_content=decode_to_str(example["context"]),
+                    metadata={
+                        "id": decode_to_str(example["id"]),
+                        "title": decode_to_str(example["title"]),
+                        "question": decode_to_str(example["question"]),
+                        "answer": decode_to_str(example["answers"]["text"][0]),
+                    },
+                )
+
+            tsds_client = TensorflowDatasets(
+                    dataset_name="mlqa/en",
+                    split_name="test",
+                    load_max_docs=MAX_DOCS,
+                    sample_to_document_function=mlqaen_example_to_document,
+                )
+
+    """  # noqa: E501
+
+    dataset_name: str = ""
+    split_name: str = "train"
+    load_max_docs: int = 100
+    sample_to_document_function: Optional[Callable[[Dict], Document]] = None
+    dataset: Any  #: :meta private:
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that the python package exists in environment."""
+        try:
+            import tensorflow  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "Could not import tensorflow python package. "
+                "Please install it with `pip install tensorflow`."
+            )
+        try:
+            import tensorflow_datasets
+        except ImportError:
+            raise ImportError(
+                "Could not import tensorflow_datasets python package. "
+                "Please install it with `pip install tensorflow-datasets`."
+            )
+        if values["sample_to_document_function"] is None:
+            raise ValueError(
+                "sample_to_document_function is None. "
+                "Please provide a function that converts a dataset sample to a Document."  # noqa: E501
+            )
+        values["dataset"] = tensorflow_datasets.load(
+            values["dataset_name"], split=values["split_name"]
+        )
+
+        return values
+
+    def lazy_load(self) -> Iterator[Document]:
+        """
```

Comment:
nits: 
* Guidelines doesn't specify exactly, but most examples i've seen do not use new line after `"""`
* New line after `Returns:`

https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings

Main question with all of these is whether sphinx is able to parse the docs correctly. (Ideally we had more uniformity in doc-strings (e.g., followed google doc style), but we haven't discussed this formally and there are no linters or contributor guidelines as far as I am aware)

Reply:
As I see sphinx is OK with all ^ options. Still, uniformity is king.